src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/config/DomainEventUrlConfig.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.config
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.context.annotation.Configuration
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DomainEventType
+
+@Configuration
+@ConfigurationProperties(prefix = "url-templates.api")
+class DomainEventUrlConfig {
+  lateinit var cas1: Map<String, String>
+  lateinit var cas2: Map<String, String>
+  lateinit var cas3: Map<String, String>
+
+  fun getUrlForDomainEventType(domainEventType: DomainEventType): String {
```

Comment:
If all templates use eventId as the param name, could this take in the event ID and return the rendered url instead of the template? This would help ensure new templates use that param in the future too

Reply:
Nice idea!